Prep for move to community repository

gparvin opened the pull request:
This creates some simple API validation (which could be enhanced more) and changes ACM specific usage to open cluster management.

I have not changed references to stolostron yet.  I'm thinking that will be a PR to do after the move to avoid breaking links now. Any feedback or PR updates are welcome.

Refs:
 - https://github.com/stolostron/backlog/issues/25068

Signed-off-by: Gus Parvin <gparvin@redhat.com>

dhaiducek commented:
I'm not so sure we should make these changes here midstream since we may want to preserve Red Hat references here in `stolostron`. It seems we should migrate the repo, fork it back, and then make the changes upstream to reflect it being upstream?

dhaiducek commented:
The changes for upstream overall look good, and that schema validator is a really nice addition! I might move it to a more generic "builds" or "scripts" folder so that we can put other scripts there in the future, though.

dhaiducek commented:
/hold

dhaiducek commented on policygenerator/policy-sets/stable/README.md:
```diff
@@ -8,4 +8,4 @@ Kustomize plugin for more information on using the policy generator.
 Policy  | Description | Prerequisites
 ------- | ----------- | -------------
-[Advanced Cluster Management Hardening](./openshift-hardening) | Applies best practices for how to harden your Advanced Cluster Management hub installation. | Needs to be deployed only to the Advanced Cluster Management hub cluster. The `PolicySet` uses cluster `Placement` and not the `PlacementRule` placement mechanism.
+[Open Cluster Management Hardening](./openshift-hardening) | Applies best practice policies for your Open Cluster Management hub installation. | Needs to be deployed only to the Open Cluster Management hub cluster. The `PolicySet` uses cluster `Placement` and not the `PlacementRule` placement mechanism.
```
```suggestion
[Open Cluster Management Hardening](./ocm-hardening) | Applies best practice policies for your Open Cluster Management hub installation. | Needs to be deployed only to the Open Cluster Management hub cluster. The `PolicySet` uses cluster `Placement` and not the `PlacementRule` placement mechanism.
```

dhaiducek replied:
Along with this, we'd need to update the name of the directory.

gparvin replied:
@dhaiducek I made changes related to renaming the hardening directory for the community policy and fixed the bad link.  I can't rename stable because we support stable and need to keep the same name until we create a `release-2.6` branch with the supported policy there.  Then we can do the rename and update the automation that tests the stable policy.

gparvin replied:
@dhaiducek As discussed I have removed the OCM specific changes for now with the goal of using magic mirror so we can have differences in upstream and midstream readme content.  Take a look to see if things look ok now.
Thanks!

mprahl commented on build/validate-policies.sh:
```diff
@@ -0,0 +1,75 @@
+
+if [ ! -f kustomize ]; then
+	echo "Downloading kustomize"
+	curl -s "https://raw.githubusercontent.com/kubernetes-sigs/kustomize/master/hack/install_kustomize.sh" | bash
```
Could you use `go install` instead? Also please use a tagged version.

gparvin replied:
Done

mprahl commented on build/validate-policies.sh:
```diff
@@ -0,0 +1,75 @@
+	curl -s "https://raw.githubusercontent.com/kubernetes-sigs/kustomize/master/hack/install_kustomize.sh" | bash
+fi
+
+if [ ! -f ~/.config/kustomize/plugin/policy.open-cluster-management.io/v1/policygenerator/PolicyGenerator ]; then
```
Could you set the `KUSTOMIZE_PLUGIN_HOME` environment variable instead of using the home directory to store the Policy Generator?

gparvin replied:
ok, I got this working with the latest changes.  Thanks for suggesting this.

mprahl replied:
@gparvin you'll need to update this if statement now to use the `KUSTOMIZE_PLUGIN_HOME` environment variable value.

gparvin commented on build/validate-policies.sh:
```diff
@@ -0,0 +1,75 @@
+	echo "Getting Schema conversion tool"
+	mkdir schemas
+	cd schemas
+	curl -o crd-schema.py https://raw.githubusercontent.com/yannh/kubeconform/master/scripts/openapi2jsonschema.py
```
Done

mprahl commented:
@gparvin could you please squash your commits?

dhaiducek reviewed (commented):
Nit: I'm still not convinced the ACM references should be updated to OCM in this PR?

gparvin commented on build/validate-policies.sh:
```diff
@@ -0,0 +1,77 @@
+
+if [ ! -f kustomize ]; then
+	echo "Installing kustomize"
+	GOBIN=$(pwd)/ GO111MODULE=on go install sigs.k8s.io/kustomize/kustomize/v4@$KUSTOMIZE_VERSION
```
Thanks, I've made updates to use GOBIN.

gparvin commented:
> Nit: I'm still not convinced the ACM references should be updated to OCM in this PR?

@dhaiducek I thought I had removed those references.  As you pointed out there were some more which I have now removed.  Thanks!

dhaiducek commented on build/validate-policies.sh:
```diff
@@ -0,0 +1,82 @@
+	chmod a+x crd-schema.py
+
+	echo "Converting CRDs to Schemas"
+	./crd-schema.py https://raw.githubusercontent.com/open-cluster-management-io/governance-policy-propagator/main/deploy/crds/policy.open-cluster-management.io_policies.yaml
+	./crd-schema.py https://raw.githubusercontent.com/open-cluster-management-io/governance-policy-propagator/main/deploy/crds/policy.open-cluster-management.io_placementbindings.yaml
+	./crd-schema.py https://raw.githubusercontent.com/open-cluster-management-io/api/main/cluster/v1beta1/0000_02_clusters.open-cluster-management.io_placements.crd.yaml
```
These should probably point to `stolostron` also. Maybe put it in a variable here and then update the workflow with `export GITHUB_ORG=${{ github.repository_owner }}`? That way, all files can be the same in all orgs.
```suggestion
	GITHUB_ORG=${GITHUB_ORG:-"open-cluster-management-io"}
	./crd-schema.py https://raw.githubusercontent.com/${GITHUB_ORG}/governance-policy-propagator/main/deploy/crds/policy.open-cluster-management.io_policies.yaml
	./crd-schema.py https://raw.githubusercontent.com/${GITHUB_ORG}/governance-policy-propagator/main/deploy/crds/policy.open-cluster-management.io_placementbindings.yaml
	./crd-schema.py https://raw.githubusercontent.com/${GITHUB_ORG}/api/main/cluster/v1beta1/0000_02_clusters.open-cluster-management.io_placements.crd.yaml
```

gparvin replied:
I made an update to use a common org.  Hopefully it will match the right one if I understand the doc correctly. THanks!

dhaiducek commented on build/validate-policies.sh:
```diff
@@ -0,0 +1,82 @@
+if [ ! -f ${KUSTOMIZE_PLUGIN_HOME}/${GENERATOR_PATH}/PolicyGenerator ]; then
+	echo "Downloading the generator"
+	PLATFORM=`uname | tr '[:upper:]' '[:lower:]'`
+	curl -L -o generator https://github.com/stolostron/policy-generator-plugin/releases/latest/download/${PLATFORM}-amd64-PolicyGenerator
```
Side note: We seriously need to get the generator upstream...

dhaiducek reviewed (commented):
Looking good! I don't recall if I said it before, but thanks for all the link updates!

I have a suggestion around dynamically setting the GitHub org.

dhaiducek reviewed (changes requested):
I didn't think about the environment variable already being available! Does removing the new `export` and using `GITHUB_REPOSITORY_OWNER` instead of `GITHUB_ORG` work?

gparvin commented:
> I didn't think about the environment variable already being available! Does removing the new `export` and using `GITHUB_REPOSITORY_OWNER` instead of `GITHUB_ORG` work?

@dhaiducek I changed it as requested but take a look and see if it's ok now.

dhaiducek reviewed (approved):
LGTM! Thanks, @gparvin!

gparvin commented:
/unhold
